src/azure-cli/azure/cli/command_modules/backup/custom_afs.py

```diff
@@ -269,12 +269,51 @@ def list_recovery_points(cmd, client, resource_group_name, vault_name, item, sta
     # Get recovery points
     recovery_points = client.list(vault_name, resource_group_name, fabric_name, container_uri, item_uri, filter_string)
     paged_recovery_points = helper.get_list_from_paged_response(recovery_points)
+    
+    if tier:
+        filtered_recovery_points = []
+        for rp in paged_recovery_points:
+            rp_tier = None
+            rp_tier_types = []
+
+            # Safely access 'additional_properties' attribute of rp.properties
+            additional_props = getattr(rp.properties, 'additional_properties', {})
+
+            # Ensure additional_props is a dictionary before using dict.get()
+            if isinstance(additional_props, dict):
+                # Safely get 'recoveryPointTierDetails' from the dictionary
+                tier_details_list = additional_props.get("recoveryPointTierDetails", [])
+        
+                # Ensure tier_details_list is a list before iterating
+                if isinstance(tier_details_list, list):
+                    for detail in tier_details_list:
+                        # Ensure each detail is a dictionary
+                        if isinstance(detail, dict):
+                            # Safely get the 'type' from each detail
+                            rp_type = detail.get("type")
+                            if rp_type:
+                                rp_tier_types.append(rp_type)
+
+            # Determine the tier of the recovery point based on the types
+            if 'InstantRP' in rp_tier_types and 'HardenedRP' in rp_tier_types:
+                rp_tier = 'SnapshotAndVaultStandard'
+            elif 'InstantRP' in rp_tier_types:
+                rp_tier = 'Snapshot'
+            elif 'HardenedRP' in rp_tier_types:
+                rp_tier = 'VaultStandard'
+            else:
+                rp_tier = 'Unknown'
+
+            # Compare with the tier parameter and add to filtered list if matched
+            if rp_tier == tier:
```

Comment:
are we just ignoring the tier parameter otherwise? We should inform the customer with a warning of some sorts, I think.